http3: flaky TestServerRequestHeaderTooLarge

```
=== RUN   TestServerRequestHeaderTooLarge/default_value
    server_test.go:381: 
        	Error Trace:	/Users/runner/work/quic-go/quic-go/http3/http3_helper_test.go:161
        	            				/Users/runner/work/quic-go/quic-go/http3/server_test.go:381
        	            				/Users/runner/work/quic-go/quic-go/http3/server_test.go:349
        	Error:      	Should be in error chain:
        	            	expected: %!q(**quic.StreamError=0x140000e8028)
        	            	in chain: "write on closed stream 0"
        	Test:       	TestServerRequestHeaderTooLarge/default_value
=== RUN   TestServerRequestHeaderTooLarge/custom_value
--- FAIL: TestServerRequestHeaderTooLarge (0.15s)
    --- FAIL: TestServerRequestHeaderTooLarge/default_value (0.15s)
    --- PASS: TestServerRequestHeaderTooLarge/custom_value (0.00s)

--- a/http3/server_test.go
+++ b/http3/server_test.go
@@ -373,13 +373,11 @@ func testServerRequestHeaderTooLarge(t *testing.T, req *http.Request, maxHeaderB
 	require.NoError(t, err)
 	_, err = str.Write(encodeRequest(t, req))
 	require.NoError(t, err)
-	require.NoError(t, str.Close())
 
 	go s.ServeQUICConn(serverConn)
 
 	expectStreamReadReset(t, str, quic.StreamErrorCode(ErrCodeFrameError))
 	expectStreamWriteReset(t, str, quic.StreamErrorCode(ErrCodeFrameError))
-
 	require.False(t, called)
 }
 

